### ISSUE TITLE
8.3.5 makes all navigation links unusable

### ISSUE THREAD

**Issue by bobek** [squidfunk/mkdocs-material#4028]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Updating to `8.3.5` renders all navigation links (left side, toc, main, search) non-reactive. They stop reacting to clicks and user cannot navigate the site.

### Expected behaviour

Links are clickable

### Actual behaviour

you cannot browse the site

### Steps to reproduce

Just build the site with `3.8.5`, going back to `3.8.4` resolves the issue.

### Package versions

```
mike==1.1.2
mkdocs-material==8.3.5
mkdocs-material-extensions==1.0.3
mkdocs-minify-plugin==0.5.0
mkdocs-redirects==1.0.4
```

`Python 3.10.5` but tested on other versions as well.

### Configuration

```yaml
site_name: Test documentation portal
theme:
  name: material
```

### System information

Tested various browsers, all behave the same

**Comment by squidfunk** [comment]
Thanks for reporting. Possibly related to the changes added in #4012.  It's not happening on our documentation site, so I suspect it's related to some specific configuration. Please provide a minimal reproducible example.

**Comment by bobek** [comment]
TBH I am slightly puzzled. I am not able to reproduce on my local machine (`mike serve` as well as `mkdocs serve`), but it manifests itself when deployed via gitlab and gitlab pages.

I am still getting the issue with rather minimal configuration
```
site_name: IP Fabric Documentation Portal
theme:
  name: material
  logo: assets/logo.png
  favicon: assets/favicon.png
  features:
    - navigation.tabs
    - search.suggest
    - search.highlight

markdown_extensions:
  - admonition
  - attr_list
  - def_list
  - pymdownx.snippets:
      check_paths: true
  - pymdownx.superfences

extra:
  version:
    provider: mike

plugins:
  - search:
  - mike:
      version_selector: true
```

I will try setup a separate repo, so we don't need to butcher our production documentation :)

**Comment by squidfunk** [comment]
Ok, I'm stupid 🥴. I accidentally left `ev.preventDefault()` in the logic of the version selector which I only included for testing that I just removed in 4f78e8195. Navigation should now work as expected. I'm very sorry for the inconvenience.

**Comment by bobek** [comment]
Dah, I have not tested static build. That would showed the behavior on my local machine as well.

> https://github.com/squidfunk/mkdocs-material/commit/4f78e8195900e96181aa12653e20ce24ed0aa672

Yup, that did the trick. Just redeployed with tip of the `master` and everything works as expected. Patch release would be great, so others don't face the same issue.

Thank you very much for your great work. Really appreciated!

**Comment by squidfunk** [comment]
Okay, perfect, many thanks for testing! I'll push out a new release today.

**Comment by squidfunk** [comment]
Released as part of 8.3.6.
